Src/DependencyCollector/Shared/Implementation/DependencyDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,306 @@
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    internal class DependencyDiagnosticSourceListener : IObserver<KeyValuePair<string, object>>, IDisposable
+    {
+        private const string ActivityNameSuffix = ".OutboundCall";
+        private const string ActivityStopNameSuffix = ".OutboundCall.Stop";
+
+        private readonly TelemetryClient client;
+        private readonly TelemetryConfiguration configuration;
+        private readonly DependencyDiagnosticSourceSubscriber subscriber;
+
+        #region fetchers
+
+        private readonly PropertyFetcher dependencyCallRequestUriFetcher = new PropertyFetcher("RequestUri");
+
+        #endregion fetchers
+
+        public DependencyDiagnosticSourceListener(TelemetryConfiguration configuration)
+        {
+            this.client = new TelemetryClient(configuration);
+            this.client.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceExtensibility + ":");
+
+            this.configuration = configuration;
+
+            this.subscriber = new DependencyDiagnosticSourceSubscriber(this);
+        }
+
+        /// <summary>
+        /// Notifies the observer that the provider has finished sending push-based notifications.
+        /// <seealso cref="IObserver{T}.OnCompleted()"/>
+        /// </summary>
+        public void OnCompleted()
+        {
+        }
+
+        /// <summary>
+        /// Notifies the observer that the provider has experienced an error condition.
+        /// <seealso cref="IObserver{T}.OnError(Exception)"/>
+        /// </summary>
+        /// <param name="error">An object that provides additional information about the error.</param>
+        public void OnError(Exception error)
+        {
+        }
+
+        public void OnNext(KeyValuePair<string, object> evnt)
+        {
+            if (evnt.Key.EndsWith(ActivityStopNameSuffix))
```

Comment:
Use `.EndsWith(ActivityStopNameSuffix, StringComparison.Ordinal)` for both speed and correctness. Otherwise you get a culture-sensitive comparison.